zetta_utils/mazepa/execution.py

```diff
@@ -57,9 +62,47 @@ def __call__(self, target: Union[Task, Flow, ExecutionState, ComparablePartial,
             checkpoint=self.checkpoint,
             checkpoint_interval_sec=self.checkpoint_interval_sec,
             raise_on_failed_checkpoint=self.raise_on_failed_checkpoint,
+            parallel_if_pool_exists=self.parallel_if_pool_exists,
         )
 
 
+@typechecked
+def execute_with_pool(
```

Comment:
This should go into `mazepa_addons`, as it's integration with other zutils stuff. It'll be clean if we try to minimize dependency of `mazepa` to `zutils`

Reply:
Should this be cut then?